src/messaging/MessageCounterSync.cpp

```diff
@@ -198,10 +198,10 @@ void MessageCounterSyncMgr::ProcessPendingGroupMsgs(NodeId peerNodeId)
         if (!entry.msgBuf.IsNull())
         {
             PacketHeader packetHeader;
-            uint16_t headerSize = 0;
 
-            if (packetHeader.Decode((entry.msgBuf)->Start(), (entry.msgBuf)->DataLength(), &headerSize) != CHIP_NO_ERROR)
+            if (packetHeader.DecodeAndConsume(entry.msgBuf) != CHIP_NO_ERROR)
             {
+                ChipLogError(ExchangeManager, "ProcessPendingGroupMsgs::Failed to decode PacketHeader");
                 break;
             }
 
```

Comment:
So if this packet header does not match the incoming nodeid, don't we need to re-encode it into the buffer?

I had forgotten that the decoding here was conditional.  We might want to decode without consuming, and then on match consume the headerSize number of bytes or something...

Please add a test that would catch this?  ;)